test/worker.py

```diff
@@ -0,0 +1,350 @@
+    """
+    return [self.master] + self.replicas + self.rdonlys
+
+  @property
+  def replica(self):
+    """Returns the first replica Tablet instance for the shard, or None."""
+    if self.replicas:
+      return self.replicas[0]
+    else:
+      return None
+
+  @property
+  def rdonly(self):
+    """Returns the first replica Tablet instance for the shard, or None."""
+    if self.rdonlys:
+      return self.rdonlys[0]
+    else:
+      return None
+
+# initial shard, covers everything
+shard_master = tablet.Tablet()
+shard_replica = tablet.Tablet()
+shard_rdonly1 = tablet.Tablet()
+
+# split shards
+# range "" - 80
+shard_0_master = tablet.Tablet()
+shard_0_replica = tablet.Tablet()
+shard_0_rdonly1 = tablet.Tablet()
+# range 80 - ""
+shard_1_master = tablet.Tablet()
+shard_1_replica = tablet.Tablet()
+shard_1_rdonly1 = tablet.Tablet()
+
+shard_tablets = ShardTablets(shard_master, [shard_replica], [shard_rdonly1])
+shard_0_tablets = ShardTablets(shard_0_master, [shard_0_replica], [shard_0_rdonly1])
+shard_1_tablets = ShardTablets(shard_1_master, [shard_1_replica], [shard_1_rdonly1])
+
+
+def setUpModule():
+  try:
+    environment.topo_server().setup()
+
+    setup_procs = [
+        shard_master.init_mysql(),
+        shard_replica.init_mysql(),
+        shard_rdonly1.init_mysql(),
+        shard_0_master.init_mysql(),
+        shard_0_replica.init_mysql(),
+        shard_0_rdonly1.init_mysql(),
+        shard_1_master.init_mysql(),
+        shard_1_replica.init_mysql(),
+        shard_1_rdonly1.init_mysql(),
+        ]
+    utils.wait_procs(setup_procs)
+  except:
+    tearDownModule()
+    raise
+
+
+def tearDownModule():
+  if utils.options.skip_teardown:
+    return
+
+  teardown_procs = [
+      shard_master.teardown_mysql(),
+      shard_replica.teardown_mysql(),
+      shard_rdonly1.teardown_mysql(),
+      shard_0_master.teardown_mysql(),
+      shard_0_replica.teardown_mysql(),
+      shard_0_rdonly1.teardown_mysql(),
+      shard_1_master.teardown_mysql(),
+      shard_1_replica.teardown_mysql(),
+      shard_1_rdonly1.teardown_mysql(),
+      ]
+  utils.wait_procs(teardown_procs, raise_on_error=False)
+
+  environment.topo_server().teardown()
+  utils.kill_sub_processes()
+  utils.remove_tmp_files()
+
+  shard_master.remove_tree()
+  shard_replica.remove_tree()
+  shard_rdonly1.remove_tree()
+  shard_0_master.remove_tree()
+  shard_0_replica.remove_tree()
+  shard_0_rdonly1.remove_tree()
+  shard_1_master.remove_tree()
+  shard_1_replica.remove_tree()
+  shard_1_rdonly1.remove_tree()
+
+
+class TestSplitCloneResiliency(unittest.TestCase):
+  """Tests that the SplitClone worker is resilient to particular failures."""
+
+  def _init_keyspace(self):
+    """Creates a `test_keyspace` keyspace, and sets a sharding key."""
+    utils.run_vtctl(['CreateKeyspace', 'test_keyspace'])
+    utils.run_vtctl(['SetKeyspaceShardingInfo', 'test_keyspace',
+                     'keyspace_id', KEYSPACE_ID_TYPE])
```

Comment:
minor nitpick, CreateKeyspace takes -sharding_column_name and -sharding_column_type so you can combine these two.